nixos/modules/system/boot/stage-2.nix

```diff
@@ -20,6 +20,11 @@ let
       pkgs.util-linux
     ] ++ lib.optional useHostResolvConf pkgs.openresolv);
     fsPackagesPath = lib.makeBinPath config.system.fsPackages;
+    inherit (config.boot) extraSystemdUnitPaths;
+    systemdUnitPathEnvVar = lib.optionalString (config.boot.extraSystemdUnitPaths != [])
+      ("SYSTEMD_UNIT_PATH="
+      + builtins.concatStringsSep ":" config.boot.extraSystemdUnitPaths
+      + ":");
```

Comment:
The trailing `:` confused me briefly before looking it up. Would it be prudent to document the reason it's necessary?

From the `systemd.unit` manpage: "If `$SYSTEMD_UNIT_PATH` ends with an empty component ("`:`"), the usual unit load path will be appended to the contents of the variable."